More App Store Version Localization APIs/models.

VaslD opened the pull request:
These APIs are now available:

- https://developer.apple.com/documentation/appstoreconnectapi/read_app_store_version_localization_information
- https://developer.apple.com/documentation/appstoreconnectapi/create_an_app_store_version_localization
- https://developer.apple.com/documentation/appstoreconnectapi/modify_an_app_store_version_localization
- https://developer.apple.com/documentation/appstoreconnectapi/delete_an_app_store_version_localization
- https://developer.apple.com/documentation/appstoreconnectapi/list_all_app_preview_sets_for_an_app_store_version_localization (NEW!)

This API is already in master:

- https://developer.apple.com/documentation/appstoreconnectapi/list_all_app_store_version_localizations_for_an_app_store_version

~~This API is not included because I do not know exactly what it does:~~

- ~~https://developer.apple.com/documentation/appstoreconnectapi/list_all_app_preview_sets_for_an_app_store_version_localization~~

~~It seems my app didn't make use of this App Store feature but I'll probably add it anyway in upcoming days.~~

VaslD commented:
Also restructured new files in Xcode project with virtual folders (Group without Folder) because there were just TOO MANY files in one place.

AvdLee reviewed (approved):
Thanks a lot for your contribution! Especially restructuring the project, that always helps! 🙂

AvdLee commented:
@VaslD I now realise this is WIP. Let me know if I need to review something again.

VaslD commented:
> @VaslD I now realise this is WIP. Let me know if I need to review something again.

I converted it to WIP because of linter check failures. They should be all fixed now.

Also added App Preview Sets related APIs so everything in this chapter is available: https://developer.apple.com/documentation/appstoreconnectapi/app_metadata/app_store_version_localizations

I did not write unit tests. I've been using these in a project so I think I can test them against Apple's production endpoints and correct any issue that may rise.

AvdLee commented:
> I did not write unit tests. I've been using these in a project so I think I can test them against Apple's production endpoints and correct any issue that may rise.

Unfortunately, that's not really something we can build forward on. All endpoints have been tested so far IIRC and if we were to depend on real-life testing, we would never write unit tests anymore, right? ;-)

It should be fairly easy to write those tests though!

AvdLee reviewed (changes requested):
Once the tests are added, it should be good to go!

VaslD commented:
> 
> 
> > I did not write unit tests. I've been using these in a project so I think I can test them against Apple's production endpoints and correct any issue that may rise.
> 
> Unfortunately, that's not really something we can build forward on. All endpoints have been tested so far IIRC and if we were to depend on real-life testing, we would never write unit tests anymore, right? ;-)
> 
> It should be fairly easy to write those tests though!

Yeah... I was trying say that I actively use this branch so I will fix any error that I encounter right away even without unit testing. I wasn't trying to use that as an excuse.

Anyway, I looked at existing tests and it seemed they only mocked request URL building but not actual HTTP exchanges (by using fake or old responses). Did I get this right? That basically I only need to verify that the URL constructed by new endpoints are correct?

AvdLee commented:
> Anyway, I looked at existing tests and it seemed they only mocked request URL building but not actual HTTP exchanges (by using fake or old responses). Did I get this right? That basically I only need to verify that the URLs constructed by new endpoints are correct?

Exactly! 🙂

AvdLee reviewed (approved):
Thanks a lot for taking the time to write the tests after asking! Looks good to me! 🚀
